addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,190 @@
+
+    @api.depends('l10n_ro_edi_document_ids', 'l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or an OK response has been received """
+        # EXTENDS 'account'
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            received_error_document = (move.l10n_ro_edi_state == 'invoice_sending' and
+                                       all(doc.state != 'invoice_sending_failed' for doc in move.l10n_ro_edi_document_ids))
+            if move.l10n_ro_edi_state == 'invoice_sent' or received_error_document:
+                move.show_reset_to_draft_button = False
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_l10n_ro_edi_state(self):
+        self.l10n_ro_edi_state = False
+        for move in self:
+            for document in move.l10n_ro_edi_document_ids.sorted():
+                if document.state in ('invoice_sending', 'invoice_sent'):
+                    move.l10n_ro_edi_state = document.state
+                    break
+
+    ################################################################################
+    # Document Shorthands & Helpers
+    ################################################################################
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sending(self, invoice, key_loading: str):
+        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending` """
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending',
+            'key_loading': key_loading,
+        })
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sending_failed(self, invoice, message: str):
+        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending_failed` """
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending_failed',
+            'message': _("Error when sending the document to the SPV:\n%s", message),
+        })
+
+    @api.model
+    def _l10n_ro_edi_create_document_invoice_sent(self, invoice, result: dict):
+        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sent` """
+        document = self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sent',
+            'key_signature': result['key_signature'],
+            'key_certificate': result['key_certificate'],
+        })
+        document.attachment_id = self.env['ir.attachment'].sudo().create({
+            'name': invoice._l10n_ro_edi_get_attachment_file_name(),
+            'raw': result['attachment_raw'],
+            'res_model': self._name,
+            'res_id': document.id,
+            'type': 'binary',
+            'mimetype': 'application/xml',
+        })
```

Comment:
```suggestion
    def _l10n_ro_edi_create_document_invoice_sending(self, key_loading: str):
        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending` """
        self.ensure_one()
        self.env['l10n_ro_edi.document'].create({
            'invoice_id': self.id,
            'state': 'invoice_sending',
            'key_loading': key_loading,
        })

    def _l10n_ro_edi_create_document_invoice_sending_failed(self, message: str):
        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending_failed` """
        self.ensure_one()
        self.env['l10n_ro_edi.document'].create({
            'invoice_id': self.id,
            'state': 'invoice_sending_failed',
            'message': _("Error when sending the document to the SPV:\n%s", message),
        })

    def _l10n_ro_edi_create_document_invoice_sent(self, result: dict):
        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sent` """
        self.ensure_one()
        document = self.env['l10n_ro_edi.document'].create({
            'invoice_id': self.id,
            'state': 'invoice_sent',
            'key_signature': result['key_signature'],
            'key_certificate': result['key_certificate'],
        })
        document.attachment_id = self.env['ir.attachment'].sudo().create({
            'name': self._l10n_ro_edi_get_attachment_file_name(),
            'raw': result['attachment_raw'],
            'res_model': self._name,
            'res_id': document.id,
            'type': 'binary',
            'mimetype': 'application/xml',
        })
```

Reply:
Also, in case of failure, create an attachment and link it to the document. The difference here is the document must be linked to the invoice only in case of invoice_sent. That way, the user can download the failed xml to debug but can retrieve the success SInvoice on the chatter when it succeeded